crates/router/src/core/payments/tokenization.rs

```diff
@@ -844,6 +945,93 @@ pub async fn save_in_locker(
+    merchant_account: &domain::MerchantAccount,
+    payment_method_data: Option<&domain::PaymentMethodData>,
+    payment_method_request: api::PaymentMethodCreate,
+    amount: Option<i64>,
+    currency: Option<storage_enums::Currency>,
+) -> RouterResult<(
+    api_models::payment_methods::PaymentMethodResponse,
+    Option<payment_methods::transformers::DataDuplicationCheck>,
+    Option<String>,
+)> {
+    let merchant_id = merchant_account.get_id();
+    let customer_id = payment_method_request
+        .customer_id
+        .clone()
+        .get_required_value("customer_id")?;
+    let network_tokenization_supported_card_networks = &state
+        .conf
+        .network_tokenization_supported_card_networks
+        .card_networks;
+
+    if let Some(domain::PaymentMethodData::Card(card)) = payment_method_data {
+        if let Some(card_network) = &card.card_network {
+            if network_tokenization_supported_card_networks.contains(card_network) {
+                if let Ok((token_response, network_token_requestor_ref_id)) =
+                    network_tokenization::make_card_network_tokenization_request(
+                        state,
+                        card,
+                        &customer_id,
+                        amount,
+                        currency,
+                    )
+                    .await
+                {
+                    // Only proceed if the tokenization was successful
+                    let card_data = api::CardDetail {
+                        card_number: token_response.token.clone(),
+                        card_exp_month: token_response.token_expiry_month.clone(),
+                        card_exp_year: token_response.token_expiry_year.clone(),
+                        card_holder_name: None,
+                        nick_name: None,
+                        card_issuing_country: None,
+                        card_network: Some(token_response.card_brand.clone()),
+                        card_issuer: None,
+                        card_type: None,
+                    };
+
+                    let (res, dc) = Box::pin(payment_methods::cards::add_card_to_locker(
+                        state,
+                        payment_method_request,
+                        &card_data,
+                        &customer_id,
+                        merchant_account,
+                        None,
+                    ))
+                    .await
+                    .change_context(errors::ApiErrorResponse::InternalServerError)
+                    .attach_printable("Add Network Token Failed")?;
+
+                    return Ok((res, dc, network_token_requestor_ref_id));
+                }
+            }
+        }
+    }
+
+    let pm_id = common_utils::generate_id(consts::ID_LENGTH, "pm");
+    let payment_method_response = api::PaymentMethodResponse {
+        merchant_id: merchant_id.clone(),
+        customer_id: Some(customer_id),
+        payment_method_id: pm_id,
```

Comment:
Shouldn't the `payment_method_id` be ideally stored in the database as well? If the ID was generated randomly and we are directly including in the response, the payment method ID cannot be used to retrieve it because we don't have that payment method ID in our database, right?